src/providers/extraction_providers/extractionProviderFactory.ts

```diff
@@ -0,0 +1,15 @@
+import { IExtractionProvider } from './extractionProvider';
+import { MockExtractionProvider } from './mockExtractionProvider';
+
+interface Config {
+	extractionProviderName: 'mock';
+}
+
+export class ExtractionProviderFactory {
+	from({ extractionProviderName }: Config): IExtractionProvider {
+		switch (extractionProviderName) {
+			case 'mock':
+				return new MockExtractionProvider();
+		}
+	}
+}
```

Comment:
I ended up changing the way I wrote the `InterpretationProviderFactory`, if that's what you based this on.

```suggestion
export const getExtractionProvider = async (config: Config): IExtractionProvider => {
	const { extractionProviderName } = config;
	switch (extractionProviderName) {
		case 'mock':
			return new MockExtractionProvider();
	}
}
```